### PR TITLE
Construct reliability calibration mem reduction

### PR DESCRIPTION
**Pull request by BelligerG**
Lowers the memory cost of construct reliability calibration table from ~3.6 GB to ~2.1 GB. Increases the time from 264.074s to 266.197s.

Description

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks for this update, @BelligerG 👍 

This looks fine to me. I've just suggested a minor docstring update.

File: improver/utilities/cube_manipulation.py [hunk 1]
```diff
@@ -253,14 +260,19 @@ def process(
                 self._check_time_bounds_ranges(cubes_in[0])
             return cubes_in[0]
 
+        if copy:
+            cube_return = lambda cube: cube.copy()
+        else:
+            cube_return = lambda cube: cube
+
```
[the remark this answers is not in the document]
Comment (BelligerG) [reply improver/utilities/cube_manipulation.py:268]: Thanks - I'll move it up to within the lambda function

File: improver/utilities/cube_manipulation.py [hunk 2]
```diff
@@ -211,7 +211,11 @@ def _check_time_bounds_ranges(cube):
                 raise ValueError(msg)
 
     def process(
-        self, cubes_in, check_time_bounds_ranges=False, slice_over_realization=False
+        self,
+        cubes_in,
+        check_time_bounds_ranges=False,
+        slice_over_realization=False,
+        copy=True,
```
Comment (bayliffe) [inline improver/utilities/cube_manipulation.py:218]: Can we have a unit test for this addition please. It is new behaviour for this plugin.

**Review by bayliffe — CHANGES_REQUESTED** [review]
Thanks for doing this. One more test please.

@gavinevans do you know if the aggregate reliability tables also needs work to reduce memory so we can reduce the [requested resources](https://github.com/MetOffice/improver_suite/blob/master/lib/python/mo/settings/app/__init__.py#L32)

**Comment by gavinevans** [comment]
@bayliffe Yes, the `aggregate_reliability_tables` step does currently also use more memory than ideal, as far as I'm aware, so if there was anything that could be improved in that plugin that would definitely be good.

**Review by bayliffe — APPROVED** [review]

**Review by gavinevans — APPROVED** [review]

**Pull request merged**